src/router/router.factory.ts

```diff
@@ -0,0 +1,14 @@
+import express from 'express';
+import { Controller } from '@/controller';
+
+export const createRouter = <TEntity>(dataController: Controller<TEntity>) => {
+  const router = express.Router();
+
+  router.get('/', dataController.getAllItems);
+  router.get('/:id', dataController.getItemById);
+  router.post('/', dataController.createItem);
+  router.put('/:id', dataController.updateItem);
```

Comment:
Is it updating an existing item or replacing the data of the existing item?
patch is used for updating and put is used when replacing the item

Reply:
This section will be refactored in the next ticket as it doesn't accurately reflect the path.  I'll keep your comment in mind.